tsdb/query_executor_test.go

```diff
@@ -0,0 +1,140 @@
+package tsdb
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdb/influxdb/influxql"
+	"github.com/influxdb/influxdb/meta"
+)
+
+func TestWritePointsAndExecuteQuery(t *testing.T) {
+	path, _ := ioutil.TempDir("", "")
+	defer os.RemoveAll(path)
+
+	store := NewStore(path)
+	err := store.Open()
+	if err != nil {
+		t.Fatalf("error opening store: %s", err.Error())
+	}
+	database := "foo"
+	retentionPolicy := "bar"
+	shardID := uint64(1)
+	store.CreateShard(database, retentionPolicy, shardID)
+
+	pt := NewPoint(
+		"cpu",
+		map[string]string{"host": "server"},
+		map[string]interface{}{"value": 1.0},
+		time.Unix(1, 2),
+	)
+
+	err = store.WriteToShard(shardID, []Point{pt})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	pt.SetTime(time.Unix(2, 3))
+	err = store.WriteToShard(shardID, []Point{pt})
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	executor := NewQueryExecutor(store)
+	executor.MetaStore = &testMetastore{}
+	executor.Stats = &fakeStats{}
+
+	ch, err := executor.ExecuteQuery(mustParseQuery("select * from cpu"), "foo", 20)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+
+	var results []*influxql.Result
+	for r := range ch {
+		results = append(results, r)
+	}
+
+	exepected := `[{"series":[{"name":"cpu","columns":["time","value"],"values":[["1970-01-01T00:00:01.000000002Z",1],["1970-01-01T00:00:02.000000003Z",1]]}]}]`
+	got := string(mustMarshalJSON(results))
+	if exepected != got {
+		t.Fatalf("exp: %s\ngot: %s", exepected, got)
+	}
+}
+
+type testMetastore struct{}
```

Comment:
It's more flexible to make a mock with settable functions instead. Then you can set the function on each test to return the data you need.

For example:

https://github.com/influxdb/influxdb/blob/d3e1163867bedb9931dcb4cbfb35960a21858743/test/meta_store.go